tools/hls-verifier/include/VerificationContext.h

```diff
@@ -9,102 +9,70 @@
-  Properties(const string &propertiesFileName);
-
-  string get(const string &key) const;
-
-private:
-  map<string, string> properties;
-};
+using namespace dynamatic;
 
-class VerificationContext {
-public:
-  VerificationContext(const string &cTbPath, const string &cFuvPath,
-                      const string &cFuvFunctionName,
-                      const string &vhdlDuvEntityName);
-
-  string getCTbPath() const;
-  string getCFuvPath() const;
-  string getCFuvFunctionName() const;
-  string getVhdlDuvEntityName() const;
-
-  string getInjectedCFuvPath() const;
-  string getCExecutablePath() const;
-  string getVhdlTestbenchPath() const;
-
-  string getBaseDir() const;
-  string getHlsVerifyDir() const;
-  string getVhdlSrcDir() const;
-
-  string getCOutDir() const;
-  string getRefOutDir() const;
-  string getVhdlOutDir() const;
-  string getInputVectorDir() const;
-
-  string getCOutPath(const CFunctionParameter &param) const;
-  string getRefOutPath(const CFunctionParameter &param) const;
-  string getVhdlOutPath(const CFunctionParameter &param) const;
-  string getInputVectorPath(const CFunctionParameter &param) const;
-
-  string getModelsimDoFileName() const;
-
-  const TokenCompare *getTokenComparator(const CFunctionParameter &param) const;
-
-  CFunction getCFuv() const;
-  vector<CFunctionParameter> getFuvOutputParams() const;
-  vector<CFunctionParameter> getFuvInputParams() const;
-  vector<CFunctionParameter> getFuvParams() const;
-
-private:
-  Properties properties;
-  CFunction fuv;
-  string cTBPath;
-  string cFUVPath;
-  string cFUVFunctionName;
-  string vhdlDUVEntityName;
-  TokenCompare defaultComparator;
-  IntegerCompare signedIntComparator;
-  IntegerCompare unsignedIntComparator;
-  FloatCompare floatComparator;
-  DoubleCompare doubleComparator;
-};
+static const std::string MODELSIM_DIR = "MGC_MSIM";
+static const std::string HDL_SRC_DIR = "HDL_SRC";
+static const std::string HDL_OUT_DIR = "HDL_OUT";
+static const std::string INPUT_VECTORS_DIR = "INPUT_VECTORS";
+static const std::string C_SOURCE_DIR = "C_SRC";
+static const std::string C_OUT_DIR = "C_OUT";
+static const std::string VSIM_SCRIPT_FILE = "simulation.do";
+static const std::string HLS_VERIFY_DIR = "HLS_VERIFY";
+static const std::string FLOAT_COMPARE_THRESHOLD = "0.00001";
+static const std::string DOUBLE_COMPARE_THRESHOLD = "0.000000000001";
+
+struct VerificationContext {
+  VerificationContext(const std::string &cFuvFunctionName,
+                      const std::string &vhdlDuvEntityName,
+                      handshake::FuncOp *funcOp)
+      : funcOp(funcOp), cFUVFunctionName(cFuvFunctionName),
+        vhdlDUVEntityName(vhdlDuvEntityName) {}
+
+  std::string getCFuvFunctionName() const {
+    if (!cFUVFunctionName.empty())
+      return cFUVFunctionName;
+    return vhdlDUVEntityName;
+  }
+
+  std::string getVhdlDuvEntityName() const {
+    if (!vhdlDUVEntityName.empty())
+      return vhdlDUVEntityName;
+    return cFUVFunctionName;
+  }
```

Comment:
I don't think this get method is used. The member variable is referenced directly.